src/main/scala/shine/DPIA/fromRise.scala

```diff
@@ -16,7 +16,9 @@ object fromRise {
     if (!r.IsClosedForm(expr)) {
       throw new Exception(s"expression is not in closed form: $expr\n\n with type ${expr.t}")
     }
-    val bnfExpr = normalize(ev).apply(betaReduction)(expr).get
+    // TODO: consider if we can remove this going forward
+    val exprWithUniqueNames = uniqueNames.enforce(expr)
+    val bnfExpr = normalize(ev).apply(betaReduction)(exprWithUniqueNames).get
```

Comment:
beta reduction can break unique names (`(fun x. x x) (fun y. y) --> (fun y. y) (fun y. y)`), maybe we should enforce unique names after normalization.